apps/dot_ext/views/authorization.py

```diff
@@ -65,16 +69,34 @@ def form_valid(self, form):
         }
         scopes = form.cleaned_data.get("scope")
         allow = form.cleaned_data.get("allow")
-        share_demographic_scopes = form.cleaned_data.get("share_demographic_scopes")
 
+        # Get beneficiary demographic scopes sharing choice
+        share_demographic_scopes = form.cleaned_data.get("share_demographic_scopes")
         set_session_auth_flow_trace_value(self.request, 'auth_share_demographic_scopes', share_demographic_scopes)
 
+        # Get scopes list available to the application
+        application_available_scopes = CapabilitiesScopes().get_available_scopes(application=application)
+
+        # Set scopes to those available to application and beneficiary demographic info choices
+        scopes = ' '.join([s for s in scopes.split(" ")
+                          if s in application_available_scopes])
+
+        # Init deleted counts
+        access_token_delete_cnt = 0
+        refresh_token_delete_cnt = 0
+
         try:
             uri, headers, body, status = self.create_authorization_response(
                 request=self.request, scopes=scopes, credentials=credentials, allow=allow
             )
         except OAuthToolkitError as error:
             response = self.error_response(error, application)
+
+            if allow is False:
+                access_token_delete_cnt, refresh_token_delete_cnt = remove_application_user_pair_tokens(application,
+                                                                                                        self.request.user)
+                remove_application_user_pair_tokens(application, self.request.user)
```

Comment:
I see the same function called twice with the same parameters, so I'm not sure what it would be doing differently the second time?

Reply:
@njdister Good catch! Forgot to remove the previous line before adding the counts. TY!  Just updated this!